File: 240129/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <climits>
using namespace std;

long long a;
long long b, c;
vector<long long> v;

int func(int x){
    long long left = 1;
    long long right = a;
    long long cnt = 0;
    while(left <= right){
        long long mid = (left + right) / 2;
        if(mid == x){
            break;
        }
        else if(mid > x){
            right = mid - 1;
            cnt++;
        }
        else{
            left = mid + 1;
            cnt++;
        }
    }
    return cnt + 1;
}

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> a >> b >> c;
    // for(int i = 1; i <= a; i++){
    //     v.push_back(i);
    // }
    long long minans = INT_MAX;
    long long maxans = 0;
    for(int i = b; i <= c; i++){
        long long ans = func(i);
        //cout << ans << ' ';
        minans = min(minans, ans);
        maxans = max(maxans, ans);
    }
    //cout << '\n';
    cout << minans << ' ' << maxans;
    return 0;
}

File: 240129/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.py
m = int(input())

a, b = map(int, input().split())

def binarySearch(target):
    l = 1
    r = m 
    count = 0
    while l <= r:
        count += 1

        mid = (l + r) // 2
        
        if mid  == target:
            return count 
        elif mid  > target:
            r = mid - 1
        else:
            l = mid + 1
    return -1
minAns = 10000000000
maxAns = -1000000000
for target in range(a, b + 1):

    count = binarySearch(target)

    minAns = min(minAns, count)
    maxAns = max(maxAns, count)
print(minAns, maxAns)